### ISSUE TITLE
Error compiling innernet 1.5.4 with Rust 1.64 on Ubuntu 20.04.2

### ISSUE THREAD

**Issue by wpyoga** [tonarino/innernet#234]
I followed the instructions at [#227](https://github.com/tonarino/innernet/issues/227#issuecomment-1191199645) and [#203](https://github.com/tonarino/innernet/issues/203#issuecomment-1069827248) but got an error while building the client:

```
   Compiling shared v1.5.4 (/home/ubuntu/custom-debian-packages/innernet/innernet-1.5.4/shared)
error[E0700]: hidden type for `impl Iterator<Item = types::CidrTree<'_>>` captures lifetime that does not appear in bounds
   --> shared/src/types.rs:263:9
    |
246 |   impl<'a> CidrTree<'a> {
    |        -- hidden type `std::iter::Map<Filter<std::slice::Iter<'a, types::Cidr>, [closure@shared/src/types.rs:265:21: 265:29]>, [closure@shared/src/types.rs:266:18: 266:26]>` captures the lifetime `'a` as defined here
...
263 | /         self.cidrs
264 | |             .iter()
265 | |             .filter(move |c| c.parent == Some(self.contents.id))
266 | |             .map(move |c| Self {
267 | |                 cidrs: self.cidrs,
268 | |                 contents: c,
269 | |             })
    | |______________^

For more information about this error, try `rustc --explain E0700`.
error: could not compile `shared` due to previous error
cargo-deb: build failed
```

I'm not that familiar with Rust, so I don't really understand what's going on there. I tried `rustc --explain E0700` but didn't understand it, either...

**Comment by mcginty** [comment]
Hey @wpyoga, wow yeah it looks like v1.5.4 does not build with the new stable (or nightly). This is fixed in https://github.com/tonarino/innernet/commit/df3b4ae78f28c0364c2fdc949bd926e18409c1d3, so you can patch that and it should build on the latest rustc, but this is also a reminder for me to push out a v1.5.5 since we need a release that can be built on the latest rust.

**Comment by mcginty** [comment]
Fixed in `v1.5.5`
